DS-Session-4-Assignment/src/question2/UniqueCharacters.java

```diff
@@ -0,0 +1,69 @@
+package question2;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * This class is used to count the number of unique characters in a string and
+ * cache to store previous string result
+ * 
+ * @author admin
+ *
+ */
+
+public class UniqueCharacters {
+
+	private Map<String,Integer> cacheMap;
+	
+	void setCache(Map<String, Integer> cacheMap) {
```

Comment:
If this method is not required, remove it.

Reply:
Done the initialization in the constructor itself